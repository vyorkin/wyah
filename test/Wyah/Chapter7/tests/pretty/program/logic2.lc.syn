Program [Decl "nor" (ELam (Var "x") (ELam (Var "y") (EApp (EApp (EApp (EVar (Var "cond")) (EVar (Var "x"))) (EVar (Var "F"))) (EApp (EVar (Var "not")) (EVar (Var "y")))))),Decl "nand" (ELam (Var "x") (ELam (Var "y") (EApp (EApp (EApp (EVar (Var "cond")) (EVar (Var "x"))) (EApp (EVar (Var "not")) (EVar (Var "y")))) (EVar (Var "T"))))),Decl "equ" (ELam (Var "p") (ELam (Var "q") (EApp (EVar (Var "not")) (EApp (EApp (EVar (Var "xor")) (EVar (Var "p"))) (EVar (Var "q")))))),Decl "xor" (ELam (Var "p") (ELam (Var "q") (EApp (EApp (EVar (Var "p")) (EApp (EApp (EVar (Var "q")) (EVar (Var "F"))) (EVar (Var "T")))) (EVar (Var "q")))))]
