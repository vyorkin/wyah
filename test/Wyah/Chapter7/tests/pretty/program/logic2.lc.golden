let nor = \x -> \y -> ((cond x) F) (not y);
let nand = \x -> \y -> ((cond x) (not y)) T;
let equ = \p -> \q -> not ((xor p) q);
let xor = \p -> \q -> (p ((q F) T)) q;